docker/builder.go

```diff
@@ -9,20 +9,20 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// ImageBuilder builds an image using the docker command-line executable.
-type ImageBuilder struct {
+// ImageBuilderTagger builds an image using the docker command-line executable.
+type ImageBuilderTagger struct {
 	exec   exec.Executor
 	dryRun bool
 }
 
-// NewImageBuilder creates a new instance of an ImageBuilder.
-func NewImageBuilder(executor exec.Executor, dryRun bool) *ImageBuilder {
-	return &ImageBuilder{executor, dryRun}
+// NewDockerImpl creates a new instance of an ImageBuilder.
+func NewDockerImpl(executor exec.Executor, dryRun bool) *ImageBuilderTagger {
```

Comment:
By convention, struct initializers should be named after the type they return, prefixed by `New`

If your initializer returns a pointer to `ImageBuilderTagger`, it may be called `NewImageBuilderTagger`

`NewDockerImpl` leads to confusion IMO (it reminds me of the Java interface/impl naming conventions)